test-utils/src/dev/clickhouse.rs

```diff
@@ -298,15 +413,22 @@ impl ClickHouseInstance {
             );
         }
         let data_path = data_dir.root_path().to_path_buf();
-        let port = wait_for_port(data_dir.log_path()).await?;
-        let address = SocketAddrV6::new(Ipv6Addr::LOCALHOST, port, 0, 0);
-        Ok(Self {
-            data_dir: Some(data_dir),
-            data_path,
-            address,
-            args,
-            child: Some(child),
-            env,
+        // NOTE: Always extract the ports, even if they're specified, to ensure
+        // that we don't return from this until the server is actually ready to
+        // accept connections.
```

Comment:
If the ports were specified and non-zero, is it worth checking that the ports we scraped match?

Reply:
Added an assertion to this effect in efb0adbb0

Reply:
I flubbed this assertion, causing every single test to fail 😭 . I'm fixing that up now.

Reply:
Should be fixed in e0f4735